File: script/fetch_worlds.rb
require 'faraday'
require 'json'
require 'date'

url = 'https://esports-api.lolesports.com/persisted/gw/getSchedule?hl=ja-JP&leagueId=98767975604431411'
header = { 'x-api-key': '0TvQnueqKa5mxJntVWt0w4LpLfEkrV1Ta8rQBb9Z' }

events = []
newer = true
until newer.nil?
  response = Faraday.get url do |req|
    req.headers = header
  end
  res = JSON.parse response.body
  events += res['data']['schedule']['events']

  newer = res['data']['schedule']['pages']['newer']
  url += "&pageToken=#{newer}"
end
events.select! { |e| e['state'] == 'unstarted' }

latest_schedule = events.map do |event|
  jst = DateTime.parse(event['startTime']).to_time + (3600 * 9)
  {
    date: jst.strftime('%F'),
    time: jst.strftime('%R'),
    home: event['match']['teams'].first['code'],
    away: event['match']['teams'].last['code']
  }

  # start_date = event['startTime'][0...10]
  # start_hour = (event['startTime'][11..12].to_i + 9).to_s.rjust(2, '0')
  # {
  #   date: start_date,
  #   time: "#{start_hour}:00",
  #   home: event['match']['teams'].first['code'],
  #   away: event['match']['teams'].last['code']
  # }
end
puts latest_schedule.to_json

File: script/fetch_worlds.js
const https = require('https');

const options = {
  host: 'esports-api.lolesports.com',
  path: '/persisted/gw/getSchedule?hl=ko-KR&leagueId=98767975604431411',
  headers: {
    'x-api-key': '0TvQnueqKa5mxJntVWt0w4LpLfEkrV1Ta8rQBb9Z'
  },
  method: 'GET'
};


https.request(options, (res) => {
  let str = '';
  res.on('data', (chunk) => {
    str += chunk;
  });

  res.on('end', () => {
    const body = JSON.parse(str);
    const events = body.data.schedule.events;

    const schedule = events.filter((event) => event.state === 'unstarted').map((event) => {
      const startTime = new Date(event.startTime);
      const date = `${startTime.getFullYear()}-${(startTime.getMonth() + 1).toString().padStart(2, '0')}-${startTime.getDate().toString().padStart(2, '0')}`;
      const time = `${startTime.getHours().toString().padStart(2, '0')}:${startTime.getMinutes().toString().padStart(2, '0')}`;

      return {
        date, time,
        home: event.match.teams[0].code,
        away: event.match.teams[1].code
      }
    });

    console.log(JSON.stringify(schedule));
  });
}).end();
